pkg/tsdb/mysql/mysql.go

```diff
@@ -150,3 +160,154 @@ func (t *mysqlQueryResultTransformer) TransformQueryError(err error) error {
 }
 
 var errQueryFailed = errors.New("query failed - please inspect Grafana server log for details")
+
+// For Driver mysql, we have the list of possible data type:
+// https://www.w3schools.com/sql/sql_datatypes.asp#:~:text=In%20MySQL%20there%20are%20three,numeric%2C%20and%20date%20and%20time.
+// Since by default, we convert all into String, we need only to handle the Numeric data types
+
+var converterList = []sqlutil.StringConverter{
+	{
+		Name:           "handle DOUBLE",
+		InputScanKind:  reflect.Struct,
+		InputTypeName:  "DOUBLE",
+		ConversionFunc: func(in *string) (*string, error) { return in, nil },
+		Replacer: &sqlutil.StringFieldReplacer{
+			OutputFieldType: data.FieldTypeNullableFloat64,
+			ReplaceFunc: func(in *string) (interface{}, error) {
+				spew.Dump(in)
+				if in == nil {
+					return nil, nil
+				}
+				v, err := strconv.ParseFloat(*in, 64)
+				if err != nil {
+					return nil, err
+				}
+				return &v, nil
+			},
+		},
+	},
+	{
+		Name:           "handle BIGINT",
+		InputScanKind:  reflect.Struct,
+		InputTypeName:  "BIGINT",
+		ConversionFunc: func(in *string) (*string, error) { return in, nil },
+		Replacer: &sqlutil.StringFieldReplacer{
+			OutputFieldType: data.FieldTypeNullableInt64,
+			ReplaceFunc: func(in *string) (interface{}, error) {
+				spew.Dump(in)
+				if in == nil {
+					return nil, nil
+				}
+				v, err := strconv.ParseInt(*in, 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				return &v, nil
+			},
+		},
+	},
+	{
+		Name:           "handle DECIMAL",
+		InputScanKind:  reflect.Slice,
+		InputTypeName:  "DECIMAL",
+		ConversionFunc: func(in *string) (*string, error) { return in, nil },
+		Replacer: &sqlutil.StringFieldReplacer{
+			OutputFieldType: data.FieldTypeNullableFloat64,
+			ReplaceFunc: func(in *string) (interface{}, error) {
+				spew.Dump(in)
+				if in == nil {
+					return nil, nil
+				}
+				v, err := strconv.ParseFloat(*in, 64)
+				if err != nil {
+					return nil, err
+				}
+				return &v, nil
+			},
+		},
+	},
+	{
+		Name:           "handle DATETIME",
```

Comment:
Feels really weird that the `sqlutil.FrameFromRows` cannot handle this. Seems like this is what causes a string to be returned, but not sure?
https://github.com/grafana/grafana-plugin-sdk-go/blob/b638fd747d60136049a428cff146ddeea1239a27/data/sqlutil/sql.go#L119

Would really like to understand if there's potentially something missing in the sqlutil package to support more types out of the box.

Reply:
I think it is because `database/sql` defers types to the driver. So what you get from a scan will depend on the database, and the driver used.

So in this case `*string` is really a synonm for []byte (since our DF doesn't have []byte).

I am curious what the types of these `InputKindScan` reflect.Struct's are. If they are things like [`database/sql` `NullTime`](https://golang.org/pkg/database/sql/#NullTime) then supporting those in `sqlutil.FrameFromRows` probably makes sense.

Reply:
point noted, we could do a clean up to sdk later on as a second step.